proxy/HostStatus.h

```diff
@@ -52,16 +54,21 @@ struct HostStatus {
   {
     static HostStatus instance;
     return instance;
-  } // return the signleton pointer.
-  void setHostStatus(const char *key, const HostStatus_t status);
-  HostStatus_t getHostStatus(const char *key);
+  } // return the signleton.
+  void setHostStatus(const char *name, const HostStatus_t status);
+  HostStatus_t getHostStatus(const char *name);
```

Comment:
`getHostStatus` reads the hash table unlocked while it could be concurrently modified by a locked writer. Is that a safe usage?